maggy/tests/test_mnemos_db.py

```
@@ -0,0 +1,135 @@
+        row = db.conn.execute("PRAGMA journal_mode").fetchone()
+        assert row[0] == "wal"
+
+
+class TestContextManager:
+    def test_closes_on_exit(self, tmp_mnemos_dir: Path):
+        with MnemosDB(tmp_mnemos_dir) as db:
+            db.insert_node(MnemoNode(
+                type="FactNode", task_id="t1", content="f",
+            ))
+        # After exit, connection should be closed
+        with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)